lib/ansible/modules/cloud/vmware/vca_vapp.py

```diff
@@ -229,6 +234,55 @@ def set_state(module):
         if not vapp.undeploy(action):
             module.fail('unable to undeploy vapp')
 
+def connect_to_network(module, vdc_name, vapp_name, network_name, network_mode):
+
+    nets = filter(lambda n: n.name == network_name, module.vca.get_networks(vdc_name))
+    assert len(nets) == 1
+    the_vdc = module.vca.get_vdc(vdc_name)
+    the_vapp = module.vca.get_vapp(the_vdc, vapp_name)
+    assert the_vapp
+    assert the_vapp.name == vapp_name
+
+    # Connect vApp
+    task = the_vapp.connect_to_network(nets[0].name, nets[0].href)
+    result = module.vca.block_until_completed(task)
+    assert result
+
+    # Connect VM
+    if(network_mode == 'pool'):
```

Comment:
@gregswift not really using irc. If I set the vm to power-on as part of the  creation, it also fails for me. I have tu run a second task for that. So I create the VM with "operation: loop" and then run it again with "operation: poweron". my play looks like:
---
# tasks file for ansible-vcd
- name: Create VM
  vca_vapp:
    state: present
    operation: noop 

    host: "{{ vcd_url }}"
    org: "{{ org_name }}"
    vdc_name: "{{ vdc_name }}"
    username: "{{ vcd_username }}"
    password: "{{ vcd_pwd }}"
    api_version: '5.6'
    service_type: vcd

    catalog_name: "{{ vcd_catalog }}"
    template_name: "{{ vcd_os_template }}"

    vapp_name: "{{ vm_name }}"
    vm_name: "{{ vm_name }}"
    vm_cpus: "{{ vm_cpu }}"
    vm_memory: "{{ vm_mem }}"
    network_name: "{{ vm_net }}"

  register: vapp

#- debug:
#        msg: "VM {{ inventory_hostname }} got IP {{vm_ip}}"

- name: PowerOn VM
  vca_vapp:
    state: present
    operation: poweron

    host: "{{ vcd_url }}"
    org: "{{ org_name }}"
    vdc_name: "{{ vdc_name }}"
    username: "{{ vcd_username }}"
    password: "{{ vcd_pwd }}"
    api_version: '5.6'
    service_type: vcd

    vapp_name: "{{ vm_name }}"